bigstore/bigstore.py

```diff
@@ -183,22 +183,35 @@ def pathnames():
                 yield sha, filename, filter == "bigstore-compress"
 
 
-def push():
-    assert_initialized()
+def pull_metadata(repository='origin'):
+    """
+    Pull metadata from repository and automatically merge it with local metadata
+
+    :param repository: git url or remote
+    """
     try:
-        sys.stderr.write("pulling bigstore metadata...")
-        g().fetch("origin", "refs/notes/bigstore:refs/notes/bigstore-remote", "--force")
+        if repository == 'origin':
+            sys.stderr.write('pulling bigstore metadata...')
+        else:
+            sys.stderr.write('pulling bigstore metadata from {}...'.format(repository))
+
+        g().fetch(repository, 'refs/notes/bigstore:refs/notes/bigstore-remote', '--force')
     except git.exc.GitCommandError:
         try:
             # Create a ref so that we can push up to the repo.
-            g().notes("--ref=bigstore", "add", "HEAD", "-m", "bigstore")
-            sys.stderr.write("done\n")
+            g().notes('--ref=bigstore', 'add', 'HEAD', '-m', 'bigstore')
+            sys.stderr.write('done\n')
         except git.exc.GitCommandError:
             # If it fails silently, an existing notes object already exists.
-            sys.stderr.write("\n")
+            sys.stderr.write('\n')
     else:
-        g().notes("--ref=bigstore", "merge", "-s", "cat_sort_uniq", "refs/notes/bigstore-remote")
-        sys.stderr.write("done\n")
+        g().notes('--ref=bigstore', 'merge', '-s', 'cat_sort_uniq', 'refs/notes/bigstore-remote')
+        sys.stderr.write('done\n')
```

Comment:
Could you put the quote changes in a separate PR? If we're going to change it we might as well do it across the board and keep it separate from other changes.